plenum/server/view_change/pre_view_change_strategies.py

```diff
@@ -0,0 +1,72 @@
+from abc import abstractmethod, ABCMeta
+from functools import partial
+
+from plenum.common.messages.fields import IntegerField
+from plenum.common.messages.message_base import MessageBase
+from plenum.common.constants import VIEW_CHANGE_START, PROPOSED_VIEW_NO, PreVCStrategies
+
+
+class ViewChangeStartMessage(MessageBase):
+    typename = VIEW_CHANGE_START
+    schema = (
+        (PROPOSED_VIEW_NO, IntegerField()),
+    )
+
+
+class PreViewChangeStrategy(metaclass=ABCMeta):
+    """Abstract class for routines before starting viewChange procedure"""
+
+    def __init__(self, view_changer):
+        self.view_changer = view_changer
+
+    @abstractmethod
+    def vc_preparation(self, proposed_view_no: int):
+        raise NotImplementedError()
+
+
+class VCStartMsgStrategy(PreViewChangeStrategy):
+    """Strategy logic:
+    - when startViewChange method was called, then put 'local' ViewChangeStart message
+    - when this message will be processed by node's nodeInBoxRouter then it will be added to replica's inBox queue
+    - when this message will be processed by replica's inBoxRouter then startViewChange method will be called
+      and view_change procedure will be continued
+    """
+
+    def vc_preparation(self, proposed_view_no: int):
+        self.set_req_handler()
+        vcs_msg = ViewChangeStartMessage(proposed_view_no)
+        nodeInBox = self.view_changer.node.nodeInBox
+        nodeInBox.append((vcs_msg, self.view_changer.node.name))
+
+    def set_req_handler(self):
+        """Handler for processing ViewChangeStart message on node's nodeInBoxRouter"""
+        def process_start_vc_msg_on_node(self, msg: ViewChangeStartMessage, frm):
+            proposed_view_no = msg.proposed_view_no
+            if proposed_view_no > self.view_changer.view_no:
+                vcs_msg = ViewChangeStartMessage(proposed_view_no)
+                self.view_changer.node.master_replica.inBox.append(vcs_msg)
+
+        """Handler for processing ViewChangeStart message on replica's inBoxRouter"""
+        def process_start_vc_msg_on_replica(self, msg: ViewChangeStartMessage):
+            proposed_view_no = msg.proposed_view_no
+            if proposed_view_no > self.node.viewNo:
+                self.node.view_changer.startViewChange(proposed_view_no, continue_vc=True)
+
+        if VIEW_CHANGE_START not in self.view_changer.node.nodeMsgRouter.routes:
```

Comment:
Why do we need to extend router of master_replica, and call `view_changer.startViewChange` from there?
Why don't extend just the router of node to call a method in view changer?
Like 
in strategy:
```
def process_start_vc_msg(view_changer, msg: ViewChangeStartMessage, frm):
            proposed_view_no = msg.proposed_view_no
            if proposed_view_no > self.view_changer.view_no:
                view_changer.node.master_replica.inBox.append(vcs_msg)
....
if VIEW_CHANGE_START not in self.view_changer.inBoxRouter.routes:
    processor = partial(process_start_vc_msg,
                                self.view_changer)
    self.view_changer.inBoxRouter.add((ViewChangeStartMessage,
                            processor))
```
in `node.py`:
```
node = self.view_changer.node
node.nodeMsgRouter.add((ViewChangeStartMessage,
                            node.sendToViewChanger))
.....
def sendToViewChanger(self, msg, frm, from_current_state: bool = False):
....
        if (isinstance(msg, ViewChangeStartMessage)):
            logger.debug("{} sending message to view changer: {}".
                         format(self, (msg, frm)))
            self.msgsToViewChanger.append((msg, frm))
```